cdap-explore/src/main/java/co/cask/cdap/hive/stream/HiveStreamInputFormat.java

```diff
@@ -134,7 +137,21 @@ public InputSplit createSplit(Path path, Path indexPath, long startTime, long en
     analyzer.clearAllowedColumnNames();
     analyzer.allowColumnName("ts");
 
-    ExprNodeGenericFuncDesc expr = Utilities.deserializeExpression(serializedExpr);
+    ExprNodeGenericFuncDesc expr;
+    // Hack to deal with the fact that older versions of Hive use
+    // Utilities.deserializeExpression(String, Configuration),
+    // whereas newer versions use Utilities.deserializeExpression(String).
+    try {
+      expr = Utilities.deserializeExpression(serializedExpr);
+    } catch (NoSuchMethodError e) {
+      try {
+        expr = (ExprNodeGenericFuncDesc) Utilities.class.getMethod(
+          "deserializeExpression", String.class, Configuration.class).invoke(null, serializedExpr, conf);
+      } catch (Exception e1) {
+        LOG.warn("Could not deserialize query predicate. A full table scan will be performed.", e1);
+        return builder.setStartTime(startTime).setEndTime(endTime);
```

Comment:
Maybe better to have the whole expression analyze logic runs inside the `try-catch` block to have one `return` at the end.